tcljava/doc/EventDeleter.3
'\"
'\" Copyright (c) 1998 Sun Microsystems, Inc.
'\"
'\" See the file "license.terms" for information on usage and redistribution
'\" of this file, and for a DISCLAIMER OF ALL WARRANTIES.
'\" 
'\" RCS: @(#) $Id: EventDeleter.3,v 1.1 1998/10/14 21:09:08 cvsadmin Exp $
'\" 
.so man.macros
.TH EventDeleter 3 1.0 Jacl "Jacl Library Procedures"
.BS
.SH INTERFACE
tcl.lang.EventDeleter -- Interface for deleting events from the
Notifier's event queue.

.SH METHODS
.sp
int \fBdeleteEvent\fR(TclEvent \fIevent\fR)

.SH ARGUMENTS
.AS TclObject "table[]" .
.AP TclEvent event
Check whether this event should be deleted from the event queue.
.BE

.SH DESCRIPTION

The EventDeleter interface is used for specifying which event should
be deleted from the event queue.  When \fBdeleteEvents\fR is called on
a Notifier instance, it will invoke the \fBdeleteEvent\fR method of
the EventDeleter argument passed to it, once for each event in the
event queue. \fBdeleteEvent\fR returns 1 for all events that should be
deleted and 0 for events that should remain in the queue.  If
\fBdeleteEvent\fR determines that an event should be removed, it
should perform appropriate clean up on the event instance.

.SH "EQUIVALENT C FUNCTIONS"
Tcl_EventDeleteProc

.SH "SEE ALSO"
TclEvent, Notifier

.SH KEYWORDS
event, notifier, event queue


tcljava/ChangeLog
2006-06-29  Mo DeJong  <mdejong@users.sourceforge.net>

        * README.jacl:
        * README.tclblend: Mention that Jacl and Tcl Blend
        are production ready and include URLs for build
        notes on the wiki.

2006-06-29  Mo DeJong  <mdejong@users.sourceforge.net>

        Update version numbers to 1.3.3.

        * configure:
        * configure.in:
        * src/pkgIndex.tcl:
        * src/jacl/tcl/lang/Interp.java:
        * src/tcljava/tcl/lang/BlendExtension.java:
        * unix/tools/vutil/current:
        * win/makefile.vc:

2006-06-29  Mo DeJong  <mdejong@users.sourceforge.net>

        * configure: Regen.
        * configure.in: Set default --prefix to /usr/local.

2006-04-05  Mo DeJong  <mdejong@users.sourceforge.net>

        * src/tclblend/tcl/lang/Interp.java (evalFile):
        Add TclBlend implementation of Interp.evalFile().

2006-04-05  Mo DeJong  <mdejong@users.sourceforge.net>

        * src/jacl/tcl/lang/ExecCmd.java (cmdProc):
        Reimplement JDK 1.3 and newer exec so that two
        threads are created. One reads stdout of the
        subprocess while the other reads stderr. This
        is needed to avoid deadlocking in waitFor()
        when a process writes a lot of stdout and stderr.
        * src/tests/tcljava/tests/exec/ExecLotsOutput.java:
        * tests/inprogress/ExecCmd.test: Add test case
        for the waitFor() exec deadlocking bug.

2005-08-08  Mo DeJong  <mdejong@users.sourceforge.net>

	Add tcljava-132-branch branch and
 	and tcljava-132-branchpoint tags.

2005-08-08  Mo DeJong  <mdejong@users.sourceforge.net>

	Add tcljava-132-2005-08-08 tag to mark tree
	state for the 1.3.2 release.

2005-07-28  Mo DeJong  <mdejong@users.sourceforge.net>

	* announce: Update.
	* docs/website/index.html: Update.

2005-07-27  Mo DeJong  <mdejong@users.sourceforge.net>

	* README.jacl: Update msys description and
	mention jaclsh.bat script.
	* README.tclblend: Mention Tcl 8.4.11 and
	Thread 2.6.2 reqs. Add msys url and
	mention jtclsh.bat script.
	* docs/website/faq.html: Update FAQ.
	* docs/website/index.html: Update homepage
	info with new 1.3.2 release.
	* docs/website/mail.html: Add mail page that
	explains how the mailing lists work.

2005-07-23  Mo DeJong  <mdejong@users.sourceforge.net>

	* src/jacl/tcl/lang/Channel.java (initOutput):
	When initializing a channel of type "file",
	set a flag to indicate that the channel
	should be sync()'ed in addition to flushing.
	* src/jacl/tcl/lang/FlushCmd.java: Print
	IO exception stack trace, this should never
	happen but if it did sa stack trace would
	be helpful.
	* src/jacl/tcl/lang/TclOutputStream.java (setSync,
	flushChannel):
	Add flag setting method so that an output
	stream can be marked so that sync() will
	be invoked when flushing.
	* tests/tcl/SocketCmd.test: Add simple socket
	test case that writes to and reads from a socket.

2005-07-21  Mo DeJong  <mdejong@users.sourceforge.net>

	* src/jacl/tcl/lang/BinaryCmd.java (FormatNumber,
	ScanNumber):
	Handle edge cases when casting from a double
	to a float. Rewrite scanning code so that
	64 bit long value is used for shifting operation.
	* src/jacl/tcl/lang/InfoCmd.java:
	* src/jacl/tcl/lang/Shell.java:
	* src/jacl/tcl/lang/SubstCmd.java:
	Invoke static method via class name.
	* src/jacl/tcl/lang/Util.java (strtod): Add
	support for "NaN", "Inf", and "-Inf" strings
 	as valid double values.
	* tests/tcl/binary.test: Enable test cases
	that were not being run because they had
	the nonPortable constraint. Java supports
	only big endian primitive types so the
	binary format and scan operations can
	just assume big endian layout. Test that
	assume little endian are not run via
	addition of the littleEndian constraint.
	Add test cases for double -> float edge cases.

2005-07-20  Mo DeJong  <mdejong@users.sourceforge.net>

	* Makefile.in: Copy new files into dist.
	* tests/tclblend/loading.test: Rework test
	so that it is only run under Win32 and
	returns the proper result.
	* win/jaclsh.bat: Removed old bat script.

2005-07-19  Mo DeJong  <mdejong@users.sourceforge.net>

	* Makefile.in: Remove old green threads code
	for "custom" target. Tcl Blend only works
	with a native threads JVM anyway. Add new
	custom2 shell which loads a JVM at runtime
	and then loads Tcl Blend like a JVM would.
	This shell make it easier to debug weird
	loading issues in Tcl Blend.
	* src/tests/tclblend/tests/JavaEval.java:
	Make the program exit(0) when an expected
	result matches the program result. Use
	positive exit status.
	* tests/tclblend/loading.test: Use the
	expected compare test in the JavaEval
	class.
	* unix/custom2.c: Add new custom shell impl
	that loads a JVM and then loads Tcl Blend into
	the JVM.

2005-07-18  Mo DeJong  <mdejong@users.sourceforge.net>

	Fix crashes when loading Tcl Blend into an
	existing JVM by invoking Tcl_FindExecutable("java")
	to init Tcl subsystems.

	* src/native/javaCmd.c: Update comment.
	* src/native/javaInterp.c (
	Java_tcl_lang_Interp_create,
	Java_tcl_lang_Interp_init,
	Java_tcl_lang_Interp_initName): Add
	TCLBLEND_DEBUG output statements.
	Add initName() method that will init the
	Tcl executable name when Tcl Blend is loaded
	from Java.
	* src/tclblend/tcl/lang/Interp.java (shlib_load,
	initName):
	Add initName() method mentioned above and
	invoke it after loading the Tcl Blend shared lib.
	* tests/tclblend/loading.test: Update tests
	to account for executable name check. Add
	test for [info nameofexecutable] in a Tcl Blend
	interp loaded into Java.

2005-07-17  Mo DeJong  <mdejong@users.sourceforge.net>

	Add startup .bat files for Jacl and Tcl Blend.
	The bat files are auto generated by configure
	with the correct paths.

	* Makefile.in: Install and test jaclsh.bat
	and jtclsh.bat. Make sure the PATH variable
	is always set with the new directories
	prepended to the PATH. Rework tricky gdb
	invocation under Win32 since the mingw
	version of gdb does not know about mingw
	style unix paths.

tcljava/src/tcljava/tcl/lang/EventDeleter.java
/*
 * EventDeleter.java --
 *
 *	Interface for deleting events in the notifier's event queue.
 *
 * Copyright (c) 1997 Sun Microsystems, Inc.
 *
 * See the file "license.terms" for information on usage and
 * redistribution of this file, and for a DISCLAIMER OF ALL
 * WARRANTIES.
 * 
 * RCS: @(#) $Id: EventDeleter.java,v 1.1 1998/10/14 21:09:14 cvsadmin Exp $
 *
 */

package tcl.lang;

/*
 * This is the interface for deleting events in the notifier's event
 * queue. It's used together with the Notifier.deleteEvents() method.
 *
 */

public interface EventDeleter {

/*
 *----------------------------------------------------------------------
 *
 * deleteEvent --
 *
 *	This method is called once for each event in the event
 *	queue. It returns 1 for all events that should be deleted and
 *	0 for events that should remain in the queue.
 *
 *	If this method determines that an event should be removed, it
 *	should perform appropriate clean up on the event object.
 *
 * Results:
 *	1 means evt should be removed from the event queue. 0
 *	otherwise.
 *
 * Side effects:
 *	After this method returns 1, the event will be removed from the
 *	event queue and will not be processed.
 *
 *----------------------------------------------------------------------
 */

public int
deleteEvent(
    TclEvent evt);		// Check whether this event should be removed.

} // end EventDeleter


